webrender/src/resource_cache.rs

```diff
@@ -801,33 +804,28 @@ impl ResourceCache {
         }
 
         self.cached_images.clear_keys(&mut self.texture_cache, |request| request.key.0 == namespace);
-        self.cached_glyphs.clear_keys(&mut self.texture_cache, |request| request.font.font_key.0 == namespace);
+        self.cached_glyphs.clear_fonts(&mut self.texture_cache, |font| font.font_key.0 == namespace);
     }
 }
 
 pub trait Resource {
-    fn texture_cache_item_id(&self) -> Option<TextureCacheItemId>;
+    fn free(&self, texture_cache: &mut TextureCache);
```

Comment:
The reverse in responsibility here (the resource doesn't know what texture cache it belongs to, but it has to know how to delete itself with any?..) is a little concerning. Perhaps, we could store some unique Id of the owning texture cache in the resource, so that we can at least assert at run time if the resource is owned by a cache that it needs to remove from. That would be fine to add in follow-ups, unless you don't see a problem with this internal API.